crates/ruff/src/rules/isort/mod.rs

```diff
@@ -930,4 +954,60 @@ mod tests {
         assert_yaml_snapshot!(snapshot, diagnostics);
         Ok(())
     }
+
+    #[test_case(Path::new("sections.py"))]
+    fn sections(path: &Path) -> Result<()> {
+        let snapshot = format!("sections_{}", path.to_string_lossy());
+        let diagnostics = test_path(
+            Path::new("isort").join(path).as_path(),
+            &Settings {
+                src: vec![test_resource_path("fixtures/isort")],
+                isort: super::settings::Settings {
+                    known_modules: KnownModules::new(
+                        vec![],
+                        vec![],
+                        vec![],
+                        vec![],
+                        FxHashMap::from_iter([("django".to_string(), vec!["django".to_string()])]),
+                    ),
+                    ..super::settings::Settings::default()
+                },
+                ..Settings::for_rule(Rule::UnsortedImports)
+            },
+        )?;
+        assert_yaml_snapshot!(snapshot, diagnostics);
```

Comment:
You'll have to change this to `assert_messages` after #3906 lands. It pretty prints the diagnostics instead of serializing them to yaml. 

```suggestion
        assert_messages!(snapshot, diagnostics);
```